src/Enable.Extensions.Queuing.AzureServiceBus/Internal/AzureServiceBusQueueClient.cs

```diff
@@ -123,19 +130,36 @@ protected override void Dispose(bool disposing)
             {
                 if (disposing)
                 {
-                    _messageReceiver.CloseAsync()
-                        .GetAwaiter()
-                        .GetResult();
-
-                    _messageSender.CloseAsync()
-                        .GetAwaiter()
-                        .GetResult();
+                    // Use Monitor to prevent race conditions
+                    Monitor.Enter(_queues);
+                    try
```

Comment:
Let's have a, empty line above this `try` statement.

Reply:
Using lock so no longer relevant.